### PR TITLE
Updated vulgarity checking to use the existing obscenity library

### LINKED ISSUE #107: Research vulgarity detecting libraries
Validation and vulgarity catchers are working. We should expand the functionality to using a vulgarity library down the line so we don't have to keep the vulgarity json longer than we need

_Originally posted by @blake-gilmore in https://github.com/UNLV-CS472-672/2024-S-GROUP4-SMTK/pull/106#pullrequestreview-1956431005_
            

### Description
Our project now has a vulgarity json to perform checks on possible curse words being used in usernames or passwords. This is not comprehensive, and a significant amount of programming would need to be done to account for as much vulgarity as we can. Instead, we should look into new vulgarity detecting libraries that have already done this work.

### Possible Solution
One possible library to begin our search:
https://www.npmjs.com/package/obscenity

### PR DESCRIPTION
**Pull request by blake-gilmore**
## Description
This PR removes the existing logic for checking vulgarity in both the login and chat message events. This is done through adding the Obscenity library to our modules and utilizing its match checks.

## Testing
All tests for the existing censorMess and vulgarLang files are still passing.

## Possible Errors
There may be differences in the vulgarity handling between the two systems, but in my tests it's stayed consistent.

### CODE REVIEW

**Review by junxjr — APPROVED** [review]
Looks good! More organized to just use the library for vulgarity checking. Great work on test cases too!

File: small-talk/components/censorMess.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
-import swearWords from './swearWords.js';
+import vulgar from './vulgarLang';
```
Comment (aiharc1) [inline small-talk/components/censorMess.js:1]: Nice work implementing the existing library

**Review by aiharc1 — APPROVED** [review]
Nice job using the obscenity library and replacing the original method of the dictionary. This is more efficient !

**Review by salazw1 — APPROVED** [review]
Good work on implementing a vulgar language library for vulgar language detection. This way, we won't have to keep a dictionary of swear words in the repository!

**Review by Redblaze74 — APPROVED** [review]
LGTM, good work finding and implementing a library into our current version. I might not have interacted with the security side of our application at all but its nice to see there exists a library that can perform what we had before and there are no issues with test coverage.

**Review by tbg510 — APPROVED** [review]
Nice that we were able to find a library for this purpose. Helps keep things simpler overall.

**Review by AlvaroH2001 — APPROVED** [review]
Everything looks great and was a very much needed change as we do not want these naughty words within the very soul of SmallTalk!

**Pull request merged**